geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexForPartitionedRegion.java

```diff
@@ -62,7 +66,7 @@ protected RepositoryManager createRepositoryManager(LuceneSerializer luceneSeria
       mapper = new HeterogeneousLuceneSerializer();
     }
     PartitionedRepositoryManager partitionedRepositoryManager =
-        new PartitionedRepositoryManager(this, mapper);
+        new PartitionedRepositoryManager(this, mapper, this.waitingThreadPoolFromDM);
```

Comment:
You mean move the call of cache.getDistributionManager().getWaitingThreadPool() to this location and not caching in in a class variable?  The PartitionedRespositoryManager will need this thread pool to be passed in (where the comment is hiliting)  The repo manager doesn't have a handle to the cache.

The luceneIndexForPartitionedRegion should probably have an explicit thread pool passed into instead of grabbing it from the cache, but the factory api would have had to change.

Reply:
I agree, the LuceneIndexForPartitionedRegion should have the thread pool passed in. But this is a step in the right direction